### ISSUE TITLE
Authentication errors print a traceback in the bodhi CLI

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#1126]
If you type the wrong password when performing an action with the bodhi CLI, it will print a traceback:

```
Traceback (most recent call last):
  File "/usr/bin/bodhi", line 9, in <module>
    load_entry_point('bodhi-client', 'console_scripts', 'bodhi')()
  File "/usr/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/vagrant/bodhi/client/__init__.py", line 620, in request
    resp = client.request(update, state)
  File "/vagrant/bodhi/client/__init__.py", line 58, in wrapper
    result = method(*args, **kwargs)
  File "/vagrant/bodhi/client/__init__.py", line 153, in request
    'csrf_token': self.csrf()})
  File "/vagrant/bodhi/client/__init__.py", line 58, in wrapper
    result = method(*args, **kwargs)
  File "/vagrant/bodhi/client/__init__.py", line 293, in csrf
    self.login(self.username, self.password)
  File "/usr/lib/python2.7/site-packages/fedora/client/openidbaseclient.py", line 289, in login
    openid_insecure=self.openid_insecure)
  File "/usr/lib/python2.7/site-packages/fedora/client/openidproxyclient.py", line 138, in openid_login
    raise AuthError(output['message'])
fedora.client.AuthError: Authentication failed
```

It should just print a nice error message instead.

**Comment by ankit01ojha** [comment]
I want to work on this can you please tell me how should I start.
Thanks.

**Comment by trishnaguha** [comment]
@ankit01ojha You need to investigate into bodhi CLI codebase https://github.com/fedora-infra/bodhi/blob/develop/bodhi/client/__init__.py

**Comment by ankit01ojha** [comment]
I am getting this error with bodhi CLI http://imgur.com/YjTT8yS . I am not able to reproduce the bug because of this.

**Comment by bowlofeggs** [comment]
@ankit01ojha it appears that you are not using the current development version of Bodhi, but bodhi 0.9 instead. If you use bodhi inside the Vagrant environment, it should be the development version. You can tell if you have the right version by using the --version flag. For example, right now I see:

```
[vagrant@localhost vagrant]$ bodhi --version
2.3.3
```

I don't believe bodhi 0.9 had a --version flag.

**Comment by ankit01ojha** [comment]
@bowlofeggs yes , its 0.9.12.1, so what should I do now?

**Comment by trishnaguha** [comment]
@ankit01ojha Try to setup the development environment with Vagrant.

**Comment by ankit01ojha** [comment]
@bowlofeggs now I am having version 2.3.3 but I am unable to reproduce the bug can you just tell me how should I reproduce this bug( I don't know how should I set password to bodhi CLI).

**Comment by bowlofeggs** [comment]
@ankit01ojha There are several actions in the bodhi CLI that require authentication. For example, you could try to create a new update as me:

```
$ bodhi updates new --user bowlofeggs ejabberd-16.09-5.fc25
```

**Comment by bowlofeggs** [comment]
The patch for this issue should be included in the upcoming 2.10.0 release:

https://github.com/fedora-infra/bodhi/pull/1744
